crates/proc-macro-test/build.rs

```diff
@@ -12,14 +12,43 @@ use std::{
 use cargo_metadata::Message;
 
 fn main() {
+    println!("cargo:rerun-if-changed=imp");
+
     let out_dir = env::var_os("OUT_DIR").unwrap();
     let out_dir = Path::new(&out_dir);
 
     let name = "proc-macro-test-impl";
     let version = "0.0.0";
+
+    let imp_dir = std::env::current_dir().unwrap().join("imp");
+
+    let staging_dir = out_dir.join("proc-macro-test-imp-staging");
+    // this'll error out if the staging dir didn't previously. using
+    // `std::fs::exists` would suffer from TOCTOU so just do our best to
+    // wip it and ignore errors.
```

Comment:
```suggestion
    // wipe it and ignore errors.
```

Is `std::fs::remove_dir_all` still buggy? https://github.com/XAMPPRocky/remove_dir_all

Reply:
Actually my comment might be nonsense 🤔 I forget which of `remove_dir_all` / `create_dir_all` has this annoying edge case I keep running into in build scripts..

Anyway if this somehow fails the build we'll get a `panic!` later on in the build script. I can remove the comment / unwrap instead if you prefer.

Reply:
Nope my instinct served me well, we can't unwrap this:

<img width="1253" alt="image" src="https://user-images.githubusercontent.com/7998310/180017302-1b4a332c-6a65-4ccc-8efb-dcc32a22a523.png">